service/src/main/java/bio/terra/drshub/services/DrsResolutionService.java

```diff
@@ -302,19 +307,51 @@ static String getObjectId(UriComponents uriComponents) {
         Optional.ofNullable(uriComponents.getPath()).orElse(""), StandardCharsets.UTF_8);
   }
 
-  private Optional<AccessMethod> getAccessMethod(DrsObject drsResponse, DrsProvider drsProvider) {
+  private Optional<AccessMethod> getAccessMethod(DrsObject drsResponse, DrsProvider drsProvider, ResolveFromEnum resolveFrom) {
+    Optional<AccessMethod> accessMethod = Optional.empty();
+    if (!isEmpty(drsResponse)) {
+      List<AccessMethod> accessMethods = getAccessMethods(drsResponse, drsProvider);
+      if (resolveFrom != null) {
+        accessMethod = getAccessMethodForCloud(accessMethods, resolveFrom);
+      }
+      // if there is no access method matching the resolveFrom cloud, or
+      // if the resolveFrom cloud was not specified, return a different access method
+      if (accessMethod.isEmpty() & !accessMethods.isEmpty()) {
```

Comment:
I think you meant to write `&&` here
```suggestion
      if (accessMethod.isEmpty() && !accessMethods.isEmpty()) {
```